src/Documentation/Getting-Started-With-Orleans/Developing-a-Client.md

```diff
@@ -3,71 +3,81 @@ layout: page
 title: Developing a Client
 ---
 
+### What Is Grain Client?
 
-Once we have our grain type implemented, we can write a client application that uses the type.
+The term "Client" or sometimes "Grain Client" is used for application code that interacts with grains but itself is not part of a grain logic.
+Client code runs outside of the cluster of [Orleans servers called "silos"](Silos.md) where grains are hosted.
+Hence, a client acts as a connector or conduit to the cluster and to all grains of the application.
 
-The following Orleans DLLs from either the `[SDK-ROOT]\Binaries\PresenceClient_ or _[SDK-ROOT]\Samples\References` directories need to be referenced in the client application project:
+![](Frontend-Cluster.png)
 
-* Orleans.dll
-* OrleansRuntimeInterfaces.dll
+Usually, clients are used on the frontend web servers to connect to an Orleans cluster that serves as a middle tier with grains executing business logic.
+In a typical setup, a frontend web server:
+* Receives a request over HTTP (or some other protocol)
+* Performs necessary authentication and authorization validation
+* Decides which grain(s) should process the request
```

Comment:
I think it'll be to confusing to use "route" here because Orleans routes requests to the grains/silos under the covers. The job of the client is to decide which grain(s) to invoke. And that's application specific logic. How the app decides would be different for different apps. Maybe "Decides which grain(s) should process the request based on the content or metadata of the request" instead?

"Controller" has a different meaning in this context. I think it is better to stay away from creating an unnecessary confusion here and be specific, not overly general. Client makes a call to a grain, not to a controller, even if conceptually the grain plays a controller role in the app design.